### ISSUE TITLE
Will Kubevirt plan to develop memory snapshot?

### ISSUE THREAD

**Issue by AmazingPangWei** [kubevirt/kubevirt#10609]
**Is your feature request related to a problem? Please describe**:
Will kubevirt support memory snapshot/restore like `virsh snapshot-create/snapshot-revert` or `virsh save/restore`?

**Describe the solution you'd like**:
I didn't find any issues about this feature, so I'm not sure if Kubevirt has plans to develop this feature. Could someone tell me some informations about this?

**Comment by agalbenus** [comment]
I'm also wondering this.. could this be supported?

**Comment by ydcool** [comment]
feature request +1

**Comment by fabiand** [comment]
Yes. There are other folks who have thought about this. @aglitke ?

**Comment by jennk123** [comment]
feature request +1

**Comment by fabiand** [comment]
The first step would be to start with a design document.

**Comment by wywself** [comment]
feature request +1

**Comment by D-XIII** [comment]
feature request +1

**Comment by iholder101** [comment]
I see that a lot of folks are interested in this.

@akalenyu Does our snapshot/restore API cover this?

**Comment by akalenyu** [comment]
> I see that a lot of folks are interested in this.
> 
> @akalenyu Does our snapshot/restore API cover this?

I know this came up before but not sure of the outcome. Maybe @ShellyKa13 knows

**Comment by ShellyKa13** [comment]
I'll start by saying the we do not do memory snapshot with out snapshot/restore API.
When I worked on the memory dump the goal at first was not clear if it was only for analyses or for actually using the dump for restore the memory (hibernation) in my research doing memory dump to use for recovery was not possible at the time, and the goal was set to do the dump only for analyses.
So to conclude currently there is no such option, and I dont believe this is in the current road map, but of course if there is a great request for this we might want to do another research in that area.
cc @aglitke

**Comment by panpan0000** [comment]
when we are familiar with VMware, the VM snapshot can include both disk state and memory RAM state.
when restore from snapshot ,the VM will be restore the previous memory state.
it helps a lot for application/programs  which takes long time to initialize , for example, LLM loading (Hmm.. `cuda-checkpoint` is another story we won't go too deep).

kvm/qemu should have similar capability as I believe like `virsh snapshot-create-as` with argument `--memspec`, which metioned in 
https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/configuring_and_managing_virtualization/creating-virtual-machine-snapshots_configuring-and-managing-virtualization#creating-virtual-machine-snapshots-entering-metadata-on-the-command-line_creating-virtual-machine-snapshots

**Comment by mohit-nagaraj** [comment]
> the VM snapshot can include both disk state and memory RAM state.

Another point to be noted, ensuring application consistency between memory and disk states could be critical. This might be worth exploring for crash-consistent restores.
